### PR TITLE
Add walingest test

### PR DESCRIPTION
**Pull request by bojanserafimov**
It's the test I've been using to generate [these](https://neondb.slack.com/archives/C04BLQ4LW7K/p1700576518753439) results. The profiler that generates the flamegraph is not included in this test. I'll send a separate PR to add it.

NOTE: This PR includes a 16MB wal file. If that's too large I can compress it down to 1MB. I don't think it's worth putting it on s3 because it complicates things, and we already have 1MB files in the repo. I'm not aware of any problems they cause.

### CODE REVIEW

File: pageserver/src/tenant.rs
Comment (koivunej) [inline pageserver/src/tenant.rs:2881]:
```suggestion
    pub(crate) async fn bootstrap_timeline(
```

File: pageserver/src/walingest.rs
Comment (koivunej) [inline pageserver/src/walingest.rs:2098]:
```suggestion
        let (tenant, ctx) = TenantHarness::create("test_ingest_real_wal")?.load().await;
```

File: pageserver/src/walingest.rs
Comment (bojanserafimov) [reply pageserver/src/walingest.rs:2109]: Yeah it's 1MB compressed. I'll do that

File: pageserver/src/walingest.rs
Comment (koivunej) [inline pageserver/src/walingest.rs:2089]: assuming this test would start to fail it would be much nicer to understand where it failed if this was a plain unit `()` returning test because then you'd have the line information whereas if this fails, you need to configure logging and then hope to catch it again.

So suggest removing the return type and just doing `unwrap` or `expect` where you'd `?`.

**Review by koivunej — COMMENTED** [review]
Discussed concern: #2592 and upcoming version bump, solution @bojanserafimov came up with: include initdb in the fixture.

File: pageserver/src/walingest.rs
Comment (koivunej) [inline pageserver/src/walingest.rs:2114]:
```suggestion
        let path = "fixtures/sk_wal_segment_from_pgbench.gz";
```

Or you'll move the `fixtures/whatever_sorry_page` to `test_data`, either is fine for me.

**Review by koivunej — APPROVED** [review]
I think this is fine for the future, and we should gain this feature in near future, hopefully before this test breaks.

Asked @arpad-m just to make sure.

**Comment by bojanserafimov** [comment]
> hopefully before this test breaks

If a change breaks this test it would also break our "recovery from WAL" playbook. Until we implement bootstrapping a timeline from compressed initdb it's better to have a test that breaks than not to have one

**Comment by arpad-m** [comment]
> Until we implement bootstrapping a timeline from compressed initdb

For that there is #5912

In general I'm fine with merging this before #5912, as I think that the postgres update PR which would break this one is probably already blocked by https://github.com/neondatabase/cloud/issues/8284 , which I want to work on after #5912 is merged.

**Pull request merged**
